devtest/app/views.py

```diff
@@ -1,38 +1,115 @@
+            'book', 'book__author').order_by('-created_at')[:5]
 
-    # Get recent reviews
-    recent_reviews = Review.objects.select_related(
-        'book', 'book__author').order_by('-created_at')[:5]
+        # Get featured books (books with highest average ratings)
+        context['featured_books'] = all_books.with_average_rating().select_related('author').prefetch_related('genres').filter(
+            review_count__gt=0
+        ).order_by('-avg_rating')[:6]
 
-    context = {
-        'total_books': total_books,
-        'total_authors': total_authors,
-        'total_genres': total_genres,
-        'total_reviews': total_reviews,
-        'featured_books': featured_books,
-        'top_authors': top_authors,
-        'recent_reviews': recent_reviews,
-    }
+        return context
 
-    return render(request, 'app/landing_page.html', context)
+
+class FullLibraryView(ListView):
+    model = Book
+    template_name = 'app/full_library.html'
+    context_object_name = 'books'  # Name for the book list in template
+    paginate_by = 12  # Paginate to 12 books per page
+
+    def get_queryset(self):
+        # Base queryset with annotations for rating and count
+        queryset = Book.objects.with_average_rating().select_related(
+            'author').prefetch_related('genres')
+        # Search by title or author
+        q = self.request.GET.get('q')
+        if q:
+            queryset = queryset.filter(
+                Q(title__icontains=q) | Q(author__name__icontains=q))
+
+        # Filter by genre
+        genre = self.request.GET.get('genre')
+        if genre:
+            queryset = queryset.filter(genres__name=genre)
+
+        # Filter by min rating
+        min_rating = self.request.GET.get('min_rating')
+        if min_rating:
+            queryset = queryset.filter(avg_rating__gte=min_rating)
+
+        # Order by highest rating first
+        return queryset.order_by('-avg_rating')
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # Genres for filter dropdown
+        context['genres'] = Genre.objects.all()
+        return context
+
+
+class BookDetailView(DetailView):
+    model = Book
+    template_name = 'app/book_detail.html'
+    context_object_name = 'book'
+
+    def get_queryset(self):
+        # Annotate the queryset with avg_rating and review_count
+        return Book.objects.with_average_rating()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['review_form'] = ReviewForm()
+        context['reviews'] = self.object.review_set.order_by('-created_at')
+        return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        form = ReviewForm(request.POST)
+        if form.is_valid():
+            review = form.save(commit=False)
+            review.book = self.object
+            review.save()
```

Comment:
I believe this can be simplified. Check to be sure, though.
```suggestion
    form.instance.book = self.object
    form.save()
```